src/main/java/org/ibp/api/java/impl/middleware/dataset/AbstractDatasetExportService.java

```diff
@@ -190,4 +213,16 @@ File getReturnFile(final Study study, final List<File> files) throws IOException
 	void setZipUtil(final ZipUtil zipUtil) {
 		this.zipUtil = zipUtil;
 	}
+
+	private MeasurementVariable buildStockIdColumn() {
```

Comment:
You can send the TemId, VariableType and factor as parameters and make it reusable @cuenyad